fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java

```diff
@@ -90,16 +90,16 @@ public void perform(final String txId, final String userPrincipal, final String
         final CreateNonRdfSourceOperationBuilder builder;
         String mimeType = contentType;
         if (externalContent == null) {
-            builder = nonRdfSourceOperationFactory.createInternalBinaryBuilder(fedoraId, requestBody);
+            builder = nonRdfSourceOperationFactory.createInternalBinaryBuilder(fedoraId.getFullId(), requestBody);
```

Comment:
I'm guessing these builder interfaces are one of the examples of places where you were thinking we might switch over to using fedoraIds? That seems like a good idea, although i can understand doing that in a followup ticket to keep down the scope of this PR

Reply:
https://jira.lyrasis.org/browse/FCREPO-3270